Detectors/EMCAL/base/include/EMCALBase/Digit.h

```diff
@@ -51,19 +52,23 @@ class Digit : public DigitBase
     return lhs;
   }
 
-  Int_t GetTower() const { return mTower; }
-  void SetTower(Int_t tower) { mTower = tower; }
+  Short_t GetTower() const { return mTower; }
+  void SetTower(Short_t tower) { mTower = tower; }
 
   Double_t GetAmplitude() const { return mAmplitude; }
   void SetAmplitude(Double_t amplitude) { mAmplitude = amplitude; }
+    
+  Int_t GetLabel() const { return mLabel; }
+  void SetLabel(Int_t label) { mLabel = label; }
 
   void PrintStream(std::ostream& stream) const;
 
  private:
   friend class boost::serialization::access;
 
-  Int_t mTower;        ///< Tower index (absolute cell ID)
+  Short_t mTower;      ///< Tower index (absolute cell ID)
   Double_t mAmplitude; ///< Amplitude
+  Int_t mLabel;        ///< Index of the corresponding entry/entries in the MC label array
```

Comment:
is this cross-reference needed?

Reply:
Yes, I think so. We need some way to connect the Digit to the MCTruth record of the hits that contributed to it. I used the TOF code as a guide and this is how they did it.